sound/soc/sof/debug.c

```diff
@@ -326,8 +326,10 @@ int snd_sof_dbg_init(struct snd_sof_dev *sdev)
 
 	debugfs_create_str("fw_path", 0444, fw_profile,
 			   (char **)&plat_data->fw_filename_prefix);
-	debugfs_create_str("fw_lib_path", 0444, fw_profile,
-			   (char **)&plat_data->fw_lib_prefix);
+	/* library path is not valid for IPC3 */
+	if (plat_data->ipc_type != SOF_IPC_TYPE_3)
```

Comment:
Why not test `plat_data->fw_lib_prefix` directly? Is it not initialized yet? If not, does this leaves a short time window of vulnerability?

Reply:
The IPC-based detection will force us to add this path for new platforms, otherwise we get a kernel oops

If we test the fw_lib_prefix, we won't be able to detect a missing setting, unless someone looks in detail at the CI logs - not something we excel at, do we?

Reply:
- If `sof_select_ipc_and_paths()` is invoked before this code, then to make sure each new platform defines `fw_lib_path` then you should had some sort of assert here and not just rely on sof-test to detect this NULL maybe, maybe not. I mean something like this:

```C
   if (plat_data->ipc_type != SOF_IPC_TYPE_3) {
       APPROPRIATE_NOT_NULL_KERNEL_ASSERT(fw_lib_prefix);
       ...
   }
```

- If `sof_select_ipc_and_paths()` is NOT invoked before this code, then there is short windows with a vulnerability. Maybe some static analyzer will find it?

Reply:
> then you should had some sort of assert here and not just rely on sof-test to detect this NULL maybe, maybe not. 

Assert "here" or maybe rather in `sof_pci_probe()` or whichever function initializes `snd_sof_pdata`. That function should be in charge of not producing garbage platform data.

Reply:
I would leave the door open for vendors to _not_ use libraries even with IPC4... The library loading taking this into account (handles wf_lib_prefix == NULL with dev_err() print).
It is still true that IPC3 does not even have support for this, so I would allocate a string "Not supported" and pass that in case of ipc_type != IPC3 && !fw_lib_path.